### ISSUE TITLE
Matplotlib creates unresponsive figures

### ISSUE THREAD

**Issue by bilderbuchi** [jupyter/notebook#1610]
Hi!

When I create the following notebook and then hit "Run all", I get an empty unresponsive blank figure window popping up:

```python
%matplotlib
```
    Using matplotlib backend: Qt4Agg
```python
import matplotlib.pyplot as plt
```
```python
f = plt.figure()
ax1 = f.add_subplot(221)
ax1.plot([0, 1], [1, 1])
```
    [<matplotlib.lines.Line2D at 0x20d94598c18>]

There's an error in the terminal running the server:
```
ERROR:tornado.application:Exception in callback functools.partial(<function wrap.<locals>.null_wrapper at 0x000001E3D97101E0>)
Traceback (most recent call last):
  File "C:\Continuum\Anaconda3\lib\site-packages\tornado\ioloop.py", line 600, in _run_callback
    ret = callback()
  File "C:\Continuum\Anaconda3\lib\site-packages\tornado\stack_context.py", line 275, in null_wrapper
    return fn(*args, **kwargs)
  File "C:\Continuum\Anaconda3\lib\site-packages\ipykernel\kernelbase.py", line 258, in enter_eventloop
    self.eventloop(self)
  File "C:\Continuum\Anaconda3\lib\site-packages\ipykernel\eventloops.py", line 73, in loop_qt4
    _notify_stream_qt(kernel, s)
  File "C:\Continuum\Anaconda3\lib\site-packages\ipykernel\eventloops.py", line 23, in _notify_stream_qt
    from IPython.core.interactiveshell import NoOpContext as context
ImportError: cannot import name 'NoOpContext'
```
I am on windows 10 with Anaconda, and 
```
ipykernel                 4.3.1                    py35_0
ipython                   5.0.0                    py35_0
ipython-notebook          4.0.4                    py35_3
ipython-qtconsole         4.0.1                    py35_4
ipython_genutils          0.1.0                    py35_0
ipywidgets                4.1.1                    py35_0
jupyter                   1.0.0                    py35_3
jupyter_client            4.3.0                    py35_0
jupyter_console           5.0.0                    py35_0
jupyter_core              4.1.0                    py35_0
matplotlib                1.5.1               np111py35_0
pyqt                      4.11.4                   py35_6
```

**Comment by bilderbuchi** [comment]
If I roll back Anaconda to get versions before a recent upgrade
```
    ipython:             4.2.0-py35_0
    jupyter_console:     4.1.1-py35_0
```
then this hanging window does not occur anymore, but I am back with an old bug I suffered from, viz. that if I hit "run all" on a notebook contain a matplotlib magic (in a separate cell!), execution permanently hangs at the magic-containing cell. If I run this cell manually, then I can hit "run all below" without problems, and the figure appears as desired.
Maybe some bugfix with unintended side-effects in either jupyter_console or ipython?

**Comment by bilderbuchi** [comment]
Issue ipython/ipython#9659 looks similar, but I *am* using the magic before import, and still the figure hangs.

**Comment by takluyver** [comment]
I suspect the traceback is significant. See ipython/ipykernel#157. We need to do a quick bugfix release of IPython, but most of the people are busy at a conference. I'll try to get to it.

**Comment by bilderbuchi** [comment]
Thanks, much appreciated!

**Comment by sem-geologist** [comment]
I just would want to say, that this does not look only matplotlib problem, but all Qt stuff (PyQt4 and PyQt5 both are affected by this bug)

**Comment by bilderbuchi** [comment]
I can confirm this is fixed with Ipython 5.1.0/ipykernel 4.4.1/notebook 4.2.2.
